test/util/range.cpp

```diff
@@ -213,6 +213,21 @@ void test_pointers()
     BOOST_CHECK(bgr::at(r2, 5) == 5);
 }
 
+void test_const_pointers()
```

Comment:
Pre-build UTF is just about speeding the tests. It's the use of the `unit-test` rule that would have reported a failing test case at the end of the `b2` execution that could have helped.

'Weird' was probably a bit to strong... What are the advantages of header-only tests (especially when the test framework is bundled and already integrated to the build system) ?